File: pset1/mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

int getPositiveNum(void);
void printPyramid(int);
void printPadding(int, int);
void printHash(int);

int main(void)
{
    // Get a positive integer from user
    int height = getPositiveNum();

    // Print the pyramid
    printPyramid(height);
}

int getPositiveNum(void)
{
    int num;
    do
    {
        num = get_int("Height: ");
    }
    while (num <= 0 || num > 8); // Reject if the number is less than or equal to 0, or greater than 8
    return num;
}

void printPyramid(int height)
{
    for (int row = 0; row < height; row++)
    {
        // Padding
        printPadding(height, row);

        //Print #
        printHash(row);

        printf("\n");
    }
}

void printPadding(int height, int row)
{
    for (int col = 0; col < (height - row - 1); col++)
    {
        printf(" ");
    }
}


void printHash(int row)
{
    for (int count = 0; count < (row + 1); count++)
    {
        printf("#");
    }
}

File: pset6/mario/more/mario.py
def main():

    # Get steps from user
    steps = get_positive_num()

    # Print the pyramid
    for row in range(1, (steps + 1)):
        print(f"{' ' * (steps - row)}{'#' * row}  {'#' * row}")


def get_positive_num():
    while True:
        steps = input("Height: ")
        if steps.isdigit():
            steps = int(steps)
            if 0 < steps < 9:   # Accept steps in range(0 to 9) exclusive
                break
    return steps


main()
